tests/helpers.py

```diff
@@ -29,3 +34,107 @@ def decode_custom_error(contract_abi, error_data, w3):
+                amount
+            }
+        }
+    """
+    vars = {"sender": sender.lower(), "receiver": receiver.lower(), "type": type}
+    request_data = {
+        'query': graphql_query,
+        'variables': vars
+    }
+    resp = requests.post(subgraph_endpoint, json=request_data)
+
+    if resp.status_code == 200:
+        data = json.loads(resp.text)["data"]["transactions"]
+        subgraph_ok, verified_transactions = verify_data_in_subgraph_response(data, "amount", amount, verified_transactions)
+        if not subgraph_ok:
+            raise Exception(f"Subgraph expected info at amount incorrect")
+        return verified_transactions
+    else:
+        raise Exception(f"Request failed with status code: {resp.status_code}")
+
+
+def check_subgraph_escrow_account(sender, receiver, totalAmountThawing, balance):
+    time.sleep(3)
+    graphql_query = """
+        query($id: String!){
+            escrowAccounts(where:{id: $id}){
+                id
+                thawEndTimestamp
+                totalAmountThawing
+                balance
+            }
+        }
+    """
+    vars = {"id": sender.lower() + "-" + receiver.lower()}
+    request_data = {
+        'query': graphql_query,
+        'variables': vars
+    }
+    resp = requests.post(subgraph_endpoint, json=request_data)
+
+    if resp.status_code == 200:
+        data = json.loads(resp.text)["data"]["escrowAccounts"]
+        # verified transactions wont be saved for escrowAccounts since this id needs to be rechecked again
+        subgraph_ok = verify_data_in_subgraph_response(data, "totalAmountThawing", totalAmountThawing, [])[0]
+        if not subgraph_ok:
+            raise Exception(f"Subgraph expected info at totalAmountThawing incorrect")
+        subgraph_ok = verify_data_in_subgraph_response(data, "balance", balance, [])[0]
+        if not subgraph_ok:
+            raise Exception(f"Subgraph expected info at balance incorrect")
+    else:
+        raise Exception(f"Request failed with status code: {resp.status_code}")
+
+
+def check_subgraph_signer(signer, authorized):
+    time.sleep(5)
+    graphql_query = """
+        query($authorized_signer: String!){
+            authorizedSigners(where :{id: $authorized_signer}) {
+                is_authorized
+            }
+        }
+    """
+    vars = {"authorized_signer": signer.lower()}
+    request_data = {
+        'query': graphql_query,
+        'variables': vars
+    }
+    resp = requests.post(subgraph_endpoint, json=request_data)
+
+    if resp.status_code == 200:
+        # Can only return one thing
+        data = json.loads(resp.text)["data"]["authorizedSigners"]
+        if data[0]["is_authorized"] != authorized:
+            raise Exception(f"Subgraph expected info at is_authorized incorrect")
+    else:
+        raise Exception(f"Request failed with status code: {resp.status_code}")
+
+
+def verify_data_in_subgraph_response(data, var_to_check, expected_value, verified_transactions = []):
```

Comment:
NEVER set `[]` as default argument: https://nikos7am.com/posts/mutable-default-arguments/

Reply:
Oof, yes forgot to remove that at the end, since its actually not necessary due to changes I did